common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -933,6 +934,50 @@ def get_courses(self, branch, **kwargs):
         # get the blocks for each course index (s/b the root)
         return self._get_structures_for_branch_and_locator(branch, self._create_course_locator, **kwargs)
 
+    @autoretry_read()
+    def get_course_summaries(self, branch, **kwargs):
+        """
+        Returns a list of `CourseSummary` which matching any given qualifiers.
+
+        qualifiers should be a dict of keywords matching the db fields or any
+        legal query for mongo to use against the active_versions collection.
+
+        Note, this is to find the current head of the named branch type.
+        To get specific versions via guid use get_course.
+
+        :param branch: the branch for which to return courses.
+        """
+        def extract_course_summary(course):
+            """
+            Extract course information from the course block for split.
+            """
+            return {
+                field: course.fields[field]
+                for field in CourseSummary.course_info_fields
+                if field in course.fields
+            }
+
+        courses_summaries = []
+        for entry, structure_info in self._get_structures_for_branch(branch, **kwargs):
```

Comment:
This method returns the entire course structure. To really optimize this, you should make sure that instead you are only returning the particular fields that you care about, and, ideally, that you're making a single query to load all of the structures, rather than one query per structure object.